--- /home/msaun/personal/rk3399/linux-rockchip-4.16/arch/arm64/boot/dts/rockchip/rk3399-firefly.dts.original	2018-06-01 23:26:21.580523917 +0930
+++ rk3399-firefly.dts	2018-06-04 15:00:33.430945525 +0930
@@ -210,8 +210,12 @@
 		regulator-always-on;
 		regulator-boot-on;
 		regulator-min-microvolt = <800000>;
-		regulator-max-microvolt = <1400000>;
+		regulator-max-microvolt = <1100000>;
 		vin-supply = <&vcc_sys>;
+
+		/* for rockchip boot on */
+		rockchip,pwm_id = <2>;
+		rockchip,pwm_voltage = <1000000>;
 	};
 };
 
@@ -458,15 +462,17 @@
 	vdd_cpu_b: regulator@40 {
 		compatible = "silergy,syr827";
 		reg = <0x40>;
-		fcs,suspend-voltage-selector = <0>;
 		regulator-name = "vdd_cpu_b";
 		regulator-min-microvolt = <712500>;
 		regulator-max-microvolt = <1500000>;
 		regulator-ramp-delay = <1000>;
+		vsel-gpios = <&gpio1 18 GPIO_ACTIVE_HIGH>;
+		fcs,suspend-voltage-selector = <0>;
 		regulator-always-on;
 		regulator-boot-on;
 		vin-supply = <&vcc_sys>;
 
+		regulator-initial-state = <3>;
 		regulator-state-mem {
 			regulator-off-in-suspend;
 		};
@@ -475,15 +481,16 @@
 	vdd_gpu: regulator@41 {
 		compatible = "silergy,syr828";
 		reg = <0x41>;
-		fcs,suspend-voltage-selector = <1>;
 		regulator-name = "vdd_gpu";
 		regulator-min-microvolt = <712500>;
 		regulator-max-microvolt = <1500000>;
 		regulator-ramp-delay = <1000>;
+		fcs,suspend-voltage-selector = <1>;
 		regulator-always-on;
 		regulator-boot-on;
 		vin-supply = <&vcc_sys>;
 
+		regulator-initial-state = <3>;
 		regulator-state-mem {
 			regulator-off-in-suspend;
 		};
